CorpusSearch/Dockerfile

```diff
@@ -0,0 +1,33 @@
+﻿FROM mcr.microsoft.com/dotnet/aspnet:8.0 AS base
+RUN apt-get update && apt-get upgrade -y && \
+    apt-get install -y nodejs \
+    npm  
+USER $APP_UID
+WORKDIR /app
+EXPOSE 8080
+EXPOSE 8081
+
+FROM mcr.microsoft.com/dotnet/sdk:8.0 AS build
+ARG BUILD_CONFIGURATION=Release
+# Install Node.js and npm
+RUN curl -fsSL https://deb.nodesource.com/setup_14.x | bash - \
+    && apt-get install -y \
+        nodejs npm \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /src
+COPY ["CorpusSearch.csproj", "CorpusSearch/"]
+WORKDIR "/src/CorpusSearch"
+RUN dotnet restore "CorpusSearch.csproj"
```

Comment:
drop a comment: stating that this is to only perform a `dotnet restore` if the `csproj` has changed